build.psm1

```diff
@@ -2508,6 +2528,95 @@ function Clear-PSRepo
     }
 }
 
+# Install PowerShell modules such as PackageManagement, PowerShellGet
+function Restore-PSModule
+{
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [string]$Name,
+
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [string]$Destination,
+
+        [string]$SourceLocation="https://powershell.myget.org/F/powershellmodule/api/v2",
+
+        [string]$RequiredVersion
+        )
+
+    $needRegister = $true
+    $RepositoryName = "mygetpsmodule"
+
+    # Check if the PackageManagement works in the base-oS or PowerShellCore
+    Get-PackageProvider -Name NuGet -ForceBootstrap -Verbose:$VerbosePreference
+
+    # Get the existing registered PowerShellGet repositories
+    $psrepos = PowerShellGet\Get-PSRepository
+
+    foreach ($repo in $psrepos)
+    {
+        if(($repo.SourceLocation -eq $SourceLocation) -or ($repo.SourceLocation.TrimEnd("/") -eq $SourceLocation.TrimEnd("/")))
+        {
+            # found a registered repository that matches the source location
+            $needRegister = $false
+            $RepositoryName = $repo.Name
+            break
+        }
+    }
+
+    if($needRegister)
+    {
+        $regVar = PowerShellGet\Get-PSRepository -Name $RepositoryName -ErrorAction SilentlyContinue
+        if($regVar)
+        {
+            PowerShellGet\UnRegister-PSRepository -Name $RepositoryName
+        }
```

Comment:
this is not needed as we are registering only if we there is no existing repo with the same source location.